### DISCUSSION TITLE
display issues after upgrade

### DISCUSSION THREAD

**Discussion by kenyapcomau** [kewisch/gdata-provider#496] [Support] [answered]
I'm starting this discussion because I'm unsure if any of the previously posted discussions and issues are applicable.

My Linux distro (openSUSE) updated TB to 102.2.2 and gdata-provider was updated to 102.0.0. I am using it to view a single Google calendar. Since then I have these problems:

1. The calendar tab doesn't show any entries in my chosen view (weekly). To be precise it shows when I unsubscribe and resubscribe to the Google calendar. But if I step to the next week, it disappears and never comes back.
2. The search panel also shows a similar behaviour, it will disappear after changing the week. In addition the preset searches such as Next 14 days, etc. don't work correctly. In particular it misses out on repeated events that fall within the time window.
3. Edit: I've also just noticed that there are no more popup notifications of events now.

Fortunately the Events panel on the Mail tab seems to work correctly, even showing how much time more to the next events in the day. Edit: overnight this has also gone blank, and only unsubscribing and resubscribing brings it back.

Deleting or editing events seems to work properly, if I can find the event in the first place.

I suspect it worked properly in the previous TB 91 because I was using the CalDAV add-on there. Unfortunately that hasn't been updated to support TB 102 so gdata-provider is my only choice.

TB seems to be a bit slower to start up, but this is before the prompt for the master password so may not be related.

One thing I should try is removing the gdata-provider add-on and adding it again. Maybe there was some previous interference from the CalDAV add-on. Edit: Reinstalling the add-on didn't make any difference.

**Comment by kenyapcomau** [comment]
Ok, I turned on logging and when I click on the synchronise button I get this:

```
Error: Same occurrence found twice, protecting you from death by recursion Ical.jsm:7300:15
Uncaught (in promise) 
Exception { name: "NS_ERROR_XPC_JAVASCRIPT_ERROR_WITH_DETAILS", message: "[JavaScript Error: \"Same occurrence found twice, protecting you from death by recursion\" {file: \"resource:///modules/calendar/Ical.jsm\" line: 7300}]'[JavaScript Error: \"Same occurrence found twice, protecting you from death by recursion\" {file: \"resource:///modules/calendar/Ical.jsm\" line: 7300}]' when calling method: [calIRecurrenceRule::getOccurrences]", result: 2153185313, filename: "resource:///modules/CalRecurrenceInfo.jsm", lineNumber: 556, columnNumber: 0, data: XPCWrappedNative_NoHelper, stack: "calculateDates@resource:///modules/CalRecurrenceInfo.jsm:556:29\ngetOccurrences@resource:///modules/CalRecurrenceInfo.jsm:645:22\n_expandOccurrences@resource:///modules/calendar/CalStorageItemModel.jsm:335:43\nstart@resource:///modules/calendar/CalStorageItemModel.jsm:201:24\n", location: XPCWrappedNative_NoHelper }
CalRecurrenceInfo.jsm:556
Error: Same occurrence found twice, protecting you from death by recursion Ical.jsm:7300:15
Uncaught (in promise) 
Exception { name: "NS_ERROR_XPC_JAVASCRIPT_ERROR_WITH_DETAILS", message: "[JavaScript Error: \"Same occurrence found twice, protecting you from death by recursion\" {file: \"resource:///modules/calendar/Ical.jsm\" line: 7300}]'[JavaScript Error: \"Same occurrence found twice, protecting you from death by recursion\" {file: \"resource:///modules/calendar/Ical.jsm\" line: 7300}]' when calling method: [calIRecurrenceRule::getOccurrences]", result: 2153185313, filename: "resource:///modules/CalRecurrenceInfo.jsm", lineNumber: 556, columnNumber: 0, data: XPCWrappedNative_NoHelper, stack: "calculateDates@resource:///modules/CalRecurrenceInfo.jsm:556:29\ngetOccurrences@resource:///modules/CalRecurrenceInfo.jsm:645:22\n_expandOccurrences@resource:///modules/calendar/CalStorageItemModel.jsm:335:43\nstart@resource:///modules/calendar/CalStorageItemModel.jsm:201:24\n", location: XPCWrappedNative_NoHelper }
CalRecurrenceInfo.jsm:556
Calendar: 
Exception { name: "NS_ERROR_XPC_JAVASCRIPT_ERROR_WITH_DETAILS", message: "[JavaScript Error: \"Same occurrence found twice, protecting you from death by recursion\" {file: \"resource:///modules/calendar/Ical.jsm\" line: 7300}]'[JavaScript Error: \"Same occurrence found twice, protecting you from death by recursion\" {file: \"resource:///modules/calendar/Ical.jsm\" line: 7300}]' when calling method: [calIRecurrenceRule::getOccurrences]", result: 2153185313, filename: "resource:///modules/CalRecurrenceInfo.jsm", lineNumber: 556, columnNumber: 0, data: XPCWrappedNative_NoHelper, stack: "calculateDates@resource:///modules/CalRecurrenceInfo.jsm:556:29\ngetOccurrences@resource:///modules/CalRecurrenceInfo.jsm:645:22\n_expandOccurrences@resource:///modules/calendar/CalStorageItemModel.jsm:335:43\nstart@resource:///modules/calendar/CalStorageItemModel.jsm:201:24\n", location: XPCWrappedNative_NoHelper }
calendar-invitations-manager.js:125
Error: Same occurrence found twice, protecting you from death by recursion Ical.jsm:7300:15
    icalrecur_iterator_next resource:///modules/calendar/Ical.jsm:7300
    getOccurrences resource:///components/calRecurrenceRule.js:105
    calculateDates resource:///modules/CalRecurrenceInfo.jsm:556
    getOccurrences resource:///modules/CalRecurrenceInfo.jsm:645
    _expandOccurrences resource:///modules/calendar/CalStorageItemModel.jsm:335
    start resource:///modules/calendar/CalStorageItemModel.jsm:201
    AsyncFunctionNext self-hosted:632
```

The above looks like recurring events are messing Thunderbird Calendar up.

**Comment by kenyapcomau** [comment, marked as answer]
Ok, following the previous error log, I went through every recurring event in my calendar and clicked on each in turn, while watching the error console. I found that a couple would trigger the messages shown above. I edited these events and resaved to the calendar, without changing any details. When all the recurring events didn't cause any errors to the console, all the desired features of the calendar work as expected. I'm guessing something changed in the internal representation of events in TB 102 from TB 91. I hope this lasts.

**Comment by leechpool-github** [comment]
I had the same problems. Calendar in TB was unusable. If unsubscribe and re-establish the link it would seem to sync ok, but changing between week and monthly views would cause things to disappear from weekly and or monthly view. I waited for a while hoping whatever was causing it might get fixed but when I found this thread I did some digging.
I ran TB in terminal and saw the errors listed above whenever it opened or I pressed Sync. I exported the calendar from Google, created a new one, into into which I imported the events file and tried to connect to that but the problems remained. I manually edited the exported 1400 plus events to separate out the recurring ones and it worked ok without the recurring ones.....
Eventually, I worked out that the issue was caused by just one event which was set to repeat / custom / monthly / 31st AND last day of month. Once I set this to only last day of month, the problem went away. :) Hope this helps some people in addition to the rest of this thread.

Given the number of events in my calendar, the thought of trying to find this was daunting and it took quite a while to manually edit the calendar exports etc. It would have helped if there was a way to get some information about which event was causing the errors. Is there anyway to do this with logging etc? (I'm not very familiar with TB because it normally all just works :) ).
